modules/stochastic_tools/src/utils/GaussianProcessHandler.C

```diff
@@ -0,0 +1,353 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "GaussianProcessHandler.h"
+#include "FEProblemBase.h"
+
+#include <petsctao.h>
+#include <petscdmda.h>
+
+#include "libmesh/petsc_vector.h"
+#include "libmesh/petsc_matrix.h"
+
+#include <math.h>
+
+namespace StochasticTools
+{
+GaussianProcessHandler::GaussianProcessHandler() : _tao_comm(MPI_COMM_SELF) {}
+
+void
+GaussianProcessHandler::initialize(const InputParameters & parameters,
+                                   const UserObjectName & covar_name,
```

Comment:
Is there a way you can just pass in a reference to the covariance object? Seems a little convoluted to get it through the fe problem from the parameters and a name.